Error when building project in docker

How to reproduce

Dockerfile:
```
# BUILD
FROM node:20-alpine as BUILD
WORKDIR /app
RUN corepack enable
RUN corepack prepare yarn@stable --activate
COPY .yarn/patches /app/.yarn/patches
COPY .yarn/releases /app/.yarn/releases
COPY .yarnrc.yml /app/
COPY ./package.json /app/
RUN yarn install --mode=skip-build && yarn dedupe
COPY . /app/
RUN yarn run build

# Final image
FROM node:20-alpine
WORKDIR /app
COPY --from=BUILD /app/dist/my-project /app/
EXPOSE 4000
CMD ["node", "server/server.mjs"]

```

docker build -t container-name .  

Error:
```
41.14 ✘ [ERROR] TS2694: Namespace '"/app/node_modules/i18next/index.v4"' has no exported member 'DefaultTReturn'. [plugin angular-compiler]                                                                               
41.14                                                                                                                                                                                                                     
41.14     node_modules/angular-i18next/lib/I18NextService.d.ts:20:79:                                                                                                                                                     
41.14       20 │ ...ionOptions | undefined): i18n.DefaultTReturn<ITranslationOptions>;
41.14          ╵                                  ~~~~~~~~~~~~~~
41.14 
41.14 
41.14 ✘ [ERROR] TS2694: Namespace '"/app/node_modules/i18next/index.v4"' has no exported member 'DefaultTReturn'. [plugin angular-compiler]
41.14 
41.14     node_modules/angular-i18next/lib/I18NextService.d.ts:21:137:
41.14       21 │ ...ionOptions | undefined): i18n.DefaultTReturn<ITranslationOptions>;
41.14          ╵                                  ~~~~~~~~~~~~~~
41.14 
41.14 
41.14 ✘ [ERROR] TS2694: Namespace '"/app/node_modules/i18next/index.v4"' has no exported member 'DefaultTReturn'. [plugin angular-compiler]
41.14 
41.14     node_modules/angular-i18next/lib/ITranslationService.d.ts:20:67:
41.14       20 │ ...?: ITranslationOptions): i18n.DefaultTReturn<ITranslationOptions>;
41.14          ╵                                  ~~~~~~~~~~~~~~
41.14 
41.14 
41.14 ✘ [ERROR] TS2694: Namespace '"/app/node_modules/i18next/index.v4"' has no exported member 'DefaultTReturn'. [plugin angular-compiler]
41.14 
41.14     node_modules/angular-i18next/lib/ITranslationService.d.ts:21:89:
41.14       21 │ ...?: ITranslationOptions): i18n.DefaultTReturn<ITranslationOptions>;
41.14          ╵                                  ~~~~~~~~~~~~~~
41.14 
41.14 
41.14 ✘ [ERROR] TS2724: '"/app/node_modules/i18next/index.v4"' has no exported member named 'TOptionsBase'. Did you mean 'TOptions'? [plugin angular-compiler]
41.14 
41.14     node_modules/angular-i18next/lib/models.d.ts:13:31:
41.14       13 │ export type PipeOptions = i18n.TOptionsBase & I18Next$Dictionary & {
41.14          ╵                                ~~~~~~~~~~~~
41.14 
41.14 
41.14 Application bundle generation failed. [40.062 seconds]
```

--- a/libs/angular-i18next/src/lib/I18NextService.ts
+++ b/libs/angular-i18next/src/lib/I18NextService.ts
@@ -56,9 +56,9 @@ export class I18NextService implements ITranslationService {
     this.i18next = i18nextInstance ?? i18nextGlobal;
   }
 
-  t(key: string | string[], options?: ITranslationOptions | undefined): i18n.DefaultTReturn<ITranslationOptions>;
-  t(key: string | string[] | (string | TemplateStringsArray)[], defaultValue: string, options?: ITranslationOptions | undefined): i18n.DefaultTReturn<ITranslationOptions>;
-  t(key: unknown, defaultValueOrOptions?: unknown, options?: unknown): i18n.DefaultTReturn<ITranslationOptions> {
+  t(key: string | string[], options?: ITranslationOptions | undefined): i18n.TFunctionReturn<i18n.Namespace, string | string[], ITranslationOptions>;
+  t(key: string | string[] | (string | TemplateStringsArray)[], defaultValue: string, options?: ITranslationOptions | undefined): i18n.TFunctionReturn<i18n.Namespace, string | string[], ITranslationOptions>;
+  t(key: unknown, defaultValueOrOptions?: unknown, options?: unknown): i18n.TFunctionReturn<i18n.Namespace, unknown, ITranslationOptions> {
     const hasDefault = !!defaultValueOrOptions && typeof defaultValueOrOptions === 'string';
 
     this.i18next.t.bind(this.i18next);

--- a/libs/angular-i18next/src/lib/ITranslationService.ts
+++ b/libs/angular-i18next/src/lib/ITranslationService.ts
@@ -5,7 +5,7 @@ import { $Dictionary as I18Next$Dictionary } from 'i18next/typescript/helpers';
 
 type Modify<T, R> = Omit<T, keyof R> & R;
 
-export type ITranslationOptions = i18n.TOptions & I18Next$Dictionary & { defaultValue?: string; };
+export type ITranslationOptions = i18n.TOptions<I18Next$Dictionary>  & { defaultValue?: string; };
 
 export type ITranslationService = Modify<Partial<i18n.i18n>, {
 
@@ -29,12 +29,12 @@ export type ITranslationService = Modify<Partial<i18n.i18n>, {
   t(
     key: string | string[],
     options?: ITranslationOptions,
-  ): i18n.DefaultTReturn<ITranslationOptions>;
+  ): i18n.TFunctionReturn<i18n.Namespace, string | string[], ITranslationOptions>;
   t(
     key: string | string[],
     defaultValue: string,
     options?: ITranslationOptions
-  ): i18n.DefaultTReturn<ITranslationOptions>;
+  ): i18n.TFunctionReturn<i18n.Namespace, string | string[], ITranslationOptions>;
 
   format: i18n.FormatFunction;
 

--- a/libs/angular-i18next/src/lib/models.ts
+++ b/libs/angular-i18next/src/lib/models.ts
@@ -6,6 +6,6 @@ export type PrependPipeOptions = {
   prependScope?: boolean;
   prependNamespace?: boolean;
 };
-export type PipeOptions = i18n.TOptionsBase & I18Next$Dictionary & { defaultValue?: string; } &
+export type PipeOptions = i18n.TOptions<I18Next$Dictionary> & { defaultValue?: string; } &
   FormatPipeOptions &
   PrependPipeOptions;

--- a/libs/angular-i18next/src/tests/mocks/MockTranslationService.ts
+++ b/libs/angular-i18next/src/tests/mocks/MockTranslationService.ts
@@ -35,7 +35,7 @@ export class MockI18NextService implements ITranslationService {
 
   t = jest.fn((key: string | string[],
     optionsOrDefault?: string | i18next.TOptions,
-    options?: i18next.TOptions): i18next.DefaultTReturn<(i18next.TOptions & { defaultValue: string; })> => {
+    options?: i18next.TOptions): i18next.TFunctionReturn<i18next.Namespace, string | string[], (i18next.TOptions & { defaultValue: string; })> => {
     if (key instanceof Array) {
       return key.length > 0 ? key[0] : '';
     }

